tests/test_converter.py

```diff
@@ -1775,10 +1816,137 @@ def test_mp_to_pauli(self, measurement_type, observable, obs_string, wire, regis
         # remaining wires are all Identity
         assert np.all([op == "I" for op in pauli_op_list])
 
+    @pytest.mark.parametrize("measurement_type", [qml.expval, qml.var])
+    @pytest.mark.parametrize("register_size", [3, 5, 7])
+    @pytest.mark.parametrize(
+        "observables, coeffs",
+        [
+            ([qml.X(0), qml.X(1)], [1, 2]),
+            ([qml.X(0), qml.X(0)], [3, -2]),
+            ([qml.X(0), qml.X(0), qml.Z(1), qml.Y(2)], [3, -2, -1, 4]),
+        ],
+    )
+    def test_mp_to_pauli_for_hamiltonian_without_dot(
+        self, measurement_type, register_size, observables, coeffs
+    ):
+        """Tests that a SparsePauliOp is created from a Hamiltonian, and that
+        it has the expected format"""
+        data = []
+        for obs in observables:
+            for pauli_term, _ in obs.pauli_rep.items():
+                pauli_string = ["I"] * register_size
+                for _, val in enumerate(pauli_term.wires):
+                    pauli_string[val] = pauli_term[val]
+                pauli_string.reverse()
+                pauli_string = ("").join(pauli_string)
+                data.append(pauli_string)
+
+        hamiltonian = qml.Hamiltonian(coeffs, observables)
+
+        obs = measurement_type(hamiltonian)
+
+        pauli_op = mp_to_pauli(obs, register_size)
+        assert isinstance(pauli_op, SparsePauliOp)
+
+        pauli_op_list = list(pauli_op.paulis.to_labels()[0])
+        # all qubits in register are accounted for
+        assert len(pauli_op_list) == register_size
+        assert pauli_op == SparsePauliOp(data=data, coeffs=coeffs).simplify()
+
+    @pytest.mark.parametrize("measurement_type", [qml.expval, qml.var])
+    @pytest.mark.parametrize("register_size", [3, 5, 7])
+    @pytest.mark.parametrize(
+        "operator",
+        [
+            (qml.X(0) + qml.X(0) + qml.Y(1) + qml.Z(2)),
+            qml.sum(X(0) + X(0) + Y(1) + Z(2)),
+            (qml.X(0) + 2 * qml.Y(1)),
+            qml.sum(X(0) + qml.s_prod(2, Y(1))),
+            qml.X(0) + qml.Y(0),
+            0.5 * X(0) + 3 * (qml.X(2) + qml.PauliY(1)),
+            0.5 * X(0) + 0.5 * qml.Y(0) - 1.5 * qml.X(0) - 0.5 * qml.Y(0),
+        ],
+    )
+    def test_mp_to_pauli_for_general_operator(
+        self,
+        measurement_type,
+        register_size,
+        operator,
+    ):
+        """Tests that a SparsePauliOp is created given any general operator that has a Pauli representation, and that it has the expected format"""
+        data = []
+        coeffs = []
+        for op in operator:
+            for pauli_term, coeff in op.pauli_rep.items():
+                pauli_string = ["I"] * register_size
+                coeffs.append(coeff)
+                for _, val in enumerate(pauli_term.wires):
+                    pauli_string[val] = pauli_term[val]
+                pauli_string.reverse()
+                pauli_string = ("").join(pauli_string)
+                data.append(pauli_string)
```

Comment:
This code is pretty much a duplication of the code used for conversion to `SparsePauliOp` above. Doing this makes us prone to weak tests. A better method would be to compute the `SparsePauliOp` for the test cases manually and put them in `@pytest.mark.parametrize` in the way you do below.

Reply:
Okay 👍